metaseq/trainer.py

```diff
@@ -966,6 +977,31 @@ def clip_grad_norm(
             skip_gradient_update_on_clip_norm=skip_gradient_update_on_clip_norm,
         )
 
+    def skip_spike(self, logging_outputs, ewm_ratio_to_skip_batch):
+        if ewm_ratio_to_skip_batch == -1:
+            return
+        loss_sum = sum(log.get("loss", 0) for log in logging_outputs)
+        sample_size = sum(log.get("sample_size", 0) for log in logging_outputs)
+        loss_t = float(loss_sum / sample_size / math.log(2))
+
+        if self._ewm_loss is None:
+            self._ewm_loss = loss_t
+
+        ewm_t_1 = self._ewm_loss
+        ewm_t = ewm(loss_t, ewm_t_1, span=9)
```

Comment:
is it worth having ewm_loss logged to our metrics?

Reply:
Might not be necessary as we can always calculate it again based on the logs? There is no use for the ewm_loss value outside of skipping. When a batch is skipped we are logging ewm_loss.